src/main.rs

```diff
@@ -1414,6 +1456,36 @@ fn cmd_diff(out: &Arc<dyn Out>, cfg: &Config, sub_args: &DiffArgs) -> Result<(),
         version2: version2.clone(),
     });
 
+    if sub_args.mode == FetchMode::Sourcegraph {
+        let url =
+            format!("https://sourcegraph.com/crates/{package}/-/compare/v{version1}...v{version2}");
+        let prompt = format!("open {url} in your browser");
+        tokio::runtime::Handle::current()
+            .block_on(prompt_criteria_eulas(
+                out,
+                cfg,
+                network.as_ref(),
+                &store,
+                package,
+                Some(version1),
+                version2,
+                prompt,
+            ))
+            .into_diagnostic()?;
+
+        match open::that(&url) {
+            Ok(()) => {
+                writeln!(out, "opened {url} in your browser");
+                return Ok(());
+            }
+            Err(_e) => {
+                return Err(miette!(
+                    "Couldn't open {url} in your browser, try --mode=local?"
+                ));
+            }
```

Comment:
I think we want to acknowledge the original cause here, because it might be useful. Can we `wrap_err` instead of throwing out the old error?